### PR TITLE
New Scroll APIs + Bug Fixes

### PR DESCRIPTION
**Pull request by sid-github**
* Added Scroll APIs that allow specifying scroll action's start point, this allows tests to get around some tricky situations where scrolling may fail because scroll action's start point is outside the scroll area (this can happen for example if scroll edges have translucent views that eat up the touches).
* Bug fixes around long press action, text type actions.
* Some test fixes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,16 @@
 language: objective-c
```
Comment (khandpur) [inline .travis.yml]: revert?

File: Demo/EarlGreyExample/README.md [hunk 2]
```diff
@@ -13,7 +13,9 @@ Getting Started
 
 Execute `pod install` from the directory where you have downloaded the source of the example, then
 `open EarlGreyExample.xcworkspace` to open the example in xcode. Ensure that your version of
-CocoaPods is upgraded to the latest 1.x version since this project uses the [1.0.0 syntax](http://blog.cocoapods.org/CocoaPods-1.0/). You can get the latest CocoaPods pre-release version by
```
Comment (khandpur) [inline Demo/EarlGreyExample/README.md]: revert?

**Comment by khandpur** [comment]
Seems like it's causing a new failure in swift test case:
Failed run:
https://travis-ci.org/google/EarlGrey/jobs/127637614
Previously passed run:
https://travis-ci.org/google/EarlGrey/jobs/126971364

File: Demo/EarlGreyExample/README.md (hunk 2 above)
Comment (sid-github) [inline Demo/EarlGreyExample/README.md]: Keeping this change as this is to keep the lines under 80chars.

File: EarlGrey.xcodeproj/xcshareddata/xcschemes/EarlGrey.xcscheme [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0720"
+   LastUpgradeVersion = "0730"
```
Comment (tirodkar) [inline EarlGrey.xcodeproj/xcshareddata/xcschemes/EarlGrey.xcscheme]: Not needed. Revert.

File: EarlGrey/Action/GREYActions.m [hunk 4]
```diff
@@ -168,7 +186,7 @@ @implementation GREYActions
       return YES;
     } else if ([element conformsToProtocol:@protocol(UITextInput)]) {
       id<GREYAction> typeAtEnd = [GREYActions grey_actionForTypeText:deleteStr
-                                                    atUITextPosition:[element endOfDocument]];
+                                               atUITextPosition:[element endOfDocument]];
```
Comment (tirodkar) [inline EarlGrey/Action/GREYActions.m]: Re-align correctly.

File: EarlGrey.xcodeproj/xcshareddata/xcschemes/EarlGrey.xcscheme (hunk 3 above)
Comment (sid-github) [inline EarlGrey.xcodeproj/xcshareddata/xcschemes/EarlGrey.xcscheme]: Done.

File: EarlGrey/Action/GREYActions.m (hunk 4 above)
Comment (sid-github) [inline EarlGrey/Action/GREYActions.m]: Done

File: Demo/EarlGreyExample/README.md (hunk 2 above)
Comment (khandpur) [inline Demo/EarlGreyExample/README.md]: lgtm

File: EarlGrey.xcodeproj/xcshareddata/xcschemes/EarlGrey.xcscheme (hunk 3 above)
Comment (khandpur) [inline EarlGrey.xcodeproj/xcshareddata/xcschemes/EarlGrey.xcscheme]: still shows up here

File: Tests/FunctionalTests/Sources/FTRScrollViewTest.m [hunk 5]
```diff
@@ -302,6 +299,53 @@ - (void)testSetContentOffsetToSameCGPointDoesNotWait{
...
+// Returns a matcher that matches if the text in the given element represents a point close to the
+// |expected| point within the given |accuracy|.
+- (GREYElementMatcherBlock *)matcherTextWithCGPointValue:(CGPoint)expected
+                                                accuracy:(CGFloat)accuracy {
+  BOOL (^matchesValueWithAccuracy)(id) = ^BOOL(id element) {
+    CGPoint actual = CGPointFromString([element text]);
+    return ABS(actual.x - expected.x) <= accuracy && ABS(actual.y - expected.y) <= accuracy;
+  };
+  return [GREYElementMatcherBlock matcherWithMatchesBlock:matchesValueWithAccuracy
+                                       descriptionBlock:^(id<GREYDescription> description) {
```
Comment (khandpur) [inline Tests/FunctionalTests/Sources/FTRScrollViewTest.m]: align colon.

File: Tests/FunctionalTests/FunctionalTests.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -868,8 +868,7 @@
 				CLANG_CXX_LIBRARY = "libc++";
 				CLANG_ENABLE_MODULES = YES;
 				CLANG_ENABLE_OBJC_ARC = YES;
-				CODE_SIGN_IDENTITY = "iPhone Developer";
```
Comment (khandpur) [inline Tests/FunctionalTests/FunctionalTests.xcodeproj/project.pbxproj]: wasn't this required for carthage?

File: EarlGrey/Common/GREYExposed.h [hunk 7]
```diff
@@ -358,5 +357,5 @@ IOHIDEventRef IOHIDEventCreateDigitizerFingerEvent(CFAllocatorRef allocator,
  *  @param enabled   A boolean that indicates if predictive typing is enabled or not.
  *  @param specifier Is unused can be @c nil.
  */
-- (void)setPredictionPreferenceValue:(NSNumber *)enabled forSpecifier:(id)specifier;
+- (void)setPredictionPreferenceValue:(id)arg1 forSpecifier:(id)arg2;
```
Comment (khandpur) [inline EarlGrey/Common/GREYExposed.h]: previous one was better, it named the arguments.

File: EarlGrey/Action/GREYActions.m (hunk 4 above)
Comment (khandpur) [inline EarlGrey/Action/GREYActions.m]: doesn't seem like it.

File: EarlGrey/Action/GREYActions.h [hunk 8]
```diff
@@ -257,9 +301,26 @@ GREY_EXPORT id<GREYAction> grey_longPressAtPointWithDuration(CGPoint point,
...
+                                                                CGFloat yStartPoint);
+
 /** Shorthand macro for GREYActions::actionForScrollToContentEdge:. */
 GREY_EXPORT id<GREYAction> grey_scrollToContentEdge(GREYContentEdge edge);
 
+/**
+ *  Shorthand macro for
+ *  GREYActions::actionForScrollToContentEdge:xOriginStartPercentage:yOriginStartPercentage:.
+ */
+GREY_EXPORT id<GREYAction> grey_scrollToContentEdgeWithStartPoint(GREYContentEdge edge,
```
Comment (khandpur) [inline EarlGrey/Action/GREYActions.h]: same here

File: EarlGrey/Action/GREYActions.h [hunk 9]
```diff
@@ -257,9 +301,26 @@ GREY_EXPORT id<GREYAction> grey_longPressAtPointWithDuration(CGPoint point,
 /** Shorthand macro for GREYActions::actionForScrollInDirection:amount:. */
 GREY_EXPORT id<GREYAction> grey_scrollInDirection(GREYDirection direction, CGFloat amount);
 
+/**
+ *  Shorthand macro for
+ *  GREYActions::actionForScrollInDirection:amount:xOriginStartPercentage:yOriginStartPercentage:.
+ */
+GREY_EXPORT id<GREYAction> grey_scrollInDirectionWithStartPoint(GREYDirection direction,
```
Comment (khandpur) [inline EarlGrey/Action/GREYActions.h]: param names for x,y should be percentages like the objc method

Comment (sid-github) [inline EarlGrey/Action/GREYActions.h]: Done.

File: EarlGrey/Action/GREYActions.h (hunk 8 above)
Comment (sid-github) [inline EarlGrey/Action/GREYActions.h]: Done.

File: Tests/FunctionalTests/FunctionalTests.xcodeproj/project.pbxproj (hunk 6 above)
Comment (sid-github) [inline Tests/FunctionalTests/FunctionalTests.xcodeproj/project.pbxproj]: Reverted.

File: EarlGrey/Common/GREYExposed.h (hunk 7 above)
Comment (sid-github) [inline EarlGrey/Common/GREYExposed.h]: Reverted.

**Pull request merged**